pandas/_libs/groupby.pyx

```diff
@@ -117,6 +121,98 @@ def group_last_object(ndarray[object, ndim=2] out,
             else:
                 out[i, j] = resx[i, j]
 
+@cython.boundscheck(False)
+@cython.wraparound(False)
+def group_rank_object(ndarray[float64_t, ndim=2] out,
+                      ndarray[object, ndim=2] values,
+                      ndarray[int64_t] labels,
+                      bint is_datetimelike, **kwargs):
+    """
+    Only transforms on axis=0
+    """
+    cdef:
+        int tiebreak
+        Py_ssize_t i, j, N, K
+        int64_t val_start=0, grp_start=0, dups=0, sum_ranks=0, grp_vals_seen=1
+        int64_t grp_na_count=0
+        ndarray[int64_t] _as
+        ndarray[object] _values
+        bint pct, ascending, keep_na
+
+    tiebreak = tiebreakers[kwargs['ties_method']]
+    ascending = kwargs['ascending']
+    pct = kwargs['pct']
+    keep_na = kwargs['na_option'] == 'keep'
+    N, K = (<object> values).shape
```

Comment:
Due to my limited understanding I am not using the K value that gets extracted here, as I couldn't figure out under what circumstance K was ever not equal to 0. Can you advise how that works or what to look at to help my comprehension?